Extensions/tag_replacer.js

```diff
@@ -44,268 +42,265 @@ XKit.extensions.tag_replacer = new Object({
+				return;
 			}
 
+			if ($("#xkit-tag-replacer-append").hasClass("selected") && replace_with === "") {
+				complain($replace_with, "Enter the tag(s) you want to append.", 'Enter a new tag (example: "I still like pandas")');
+				return;
+			}
+
+			XKit.extensions.tag_replacer.start(url, replace, replace_with);
+
 		});
 
 	},
 
-	page: 0,
 	url: "",
-	t_replace: "",
-	t_with: "",
-	p_array: [],
-	p_array_index: 0,
+	replace: "",
+	replace_with: "",
+	post_ids: [],
+	post_count: 0,
 	append_mode: false,
 	case_sensitive: false,
 	success_count: 0,
 	fail_count: 0,
 
-	start: function(url, t_replace, t_with, append_mode) {
-
-		console.log("replace " + t_replace + " with " + t_with + " on " + url);
+	start: function(url, replace, replace_with) {
+
+		Object.assign(this, {
+			url: url,
+			replace: replace,
+			replace_with: replace_with,
+			post_ids: [],
+			post_count: 0,
+			success_count: 0,
+			fail_count: 0,
+			append_mode: $("#xkit-tag-replacer-append").hasClass("selected"),
+			case_sensitive: $("#xkit-tag-replacer-case-sensitive").hasClass("selected")
+		});
 
-		XKit.extensions.tag_replacer.page = 0;
-		XKit.extensions.tag_replacer.url = url;
-		XKit.extensions.tag_replacer.t_replace = t_replace;
-		XKit.extensions.tag_replacer.t_with = t_with;
-		XKit.extensions.tag_replacer.p_array = [];
-		XKit.extensions.tag_replacer.p_array_index = 0;
-		XKit.extensions.tag_replacer.success_count = 0;
-		XKit.extensions.tag_replacer.fail_count = 0;
-		XKit.extensions.tag_replacer.append_mode = append_mode;
-		XKit.extensions.tag_replacer.case_sensitive = $("#xkit-tag-replacer-case-sensitive").hasClass("selected");
+		XKit.window.show(
+			"Working...",
 
-		XKit.extensions.tag_replacer.next();
+			"Tag Replacer is trying to find posts with the tag you've entered, please wait. This might take a while." +
+			'<div id="xkit-tag-replacer-loaded">Initializing...</div>',
 
-		XKit.window.show("Working...", "Tag Replacer is trying to find posts with the tag you've entered, please wait. This might take a while.<div id=\"xkit-tag-replacer-progress\">Initializing...</div>", "info");
+			"info"
+		);
 
+		this.fetch_posts();
 	},
 
-	show_error: function(message) {
+	fetch_posts: function(page = 0) {
+		GM_xmlhttpRequest({
+			method: "GET",
+			url: `https://api.tumblr.com/v2/blog/${this.url}/posts?` + $.param({
+				"api_key": XKit.api_key,
+				"tag": this.replace,
+				"limit": 50,
+				"offset": page * 50
+			}),
+			onload: response => {
+				let data = JSON.parse(response.responseText);
+
+				if (data.response.posts.length === 0) {
+					this.start_replace();
+					return;
+				}
+
+				data.response.posts.forEach(post => this.post_ids.push(post.id));
+				$("#xkit-tag-replacer-loaded").html(`Loaded ${this.post_ids.length} tagged posts...`);
 
-		XKit.window.show("Tag Replacer encountered an error", message, "error", "<div class=\"xkit-button default\" id=\"xkit-close-message\">OK</div>");
+				this.fetch_posts(++page);
```

Comment:
This feels unnecessarily terse compared to `page + 1`

Reply:
i like it too much im sorry

Reply:
agree w/ once, this is officially Too Cute

Reply:
fine :(

Reply:
(the main problem is that ++ implies you also care about mutating the variable, which you don't)